src/network/network.cpp

```diff
@@ -446,6 +446,38 @@ static void CheckPauseOnJoin()
 	CheckPauseHelper(NetworkHasJoiningClient(), PM_PAUSED_JOIN);
 }
 
+/**
+ * Converts a string to ip/port/company
+ * Converts a string to ip/port
+ *  Format: IP:port
+ *
+ * connection_string will be re-terminated to separate out the hostname, port will
+ * be set to the port strings given by the user, inside the memory area originally
+ * occupied by connection_string.
+ */
+void ParseConnectionString(const char **port, char *connection_string)
+{
+	bool ipv6 = (strchr(connection_string, ':') != strrchr(connection_string, ':'));
+	char *p;
+	for (p = connection_string; *p != '\0'; p++) {
```

Comment:
```suggestion
	for (char *p = connection_string; *p != '\0'; p++) {
```
no need for C89 style here

Reply:
It is a copy/paste from existing code .. argh .. this never stops hunting us, does it?